### DISCUSSION TITLE
Proxying Umami via Nginx

### DISCUSSION THREAD

**Discussion by mcnaveen** [umami-software/umami#2348] [Show and tell]
![image](https://github.com/umami-software/umami/assets/8493007/44cbe3e0-26db-4437-bfb6-850ec29e7453)

For some reason, if you want to hide the umami URL under the main domain URL. You can do this.

Here change all the `umami.example.com` to your umami instance URL.

> This has to be done in the client nginx config, not in the umami instance.

Nginx Config:
```
   set $umami_script_url https://umami.example.com/script.js; # Here
   set $umami_event_url https://umami.example.com/api/send;  # Here

    location = /js/script.js {
        proxy_pass $umami_script_url;
        proxy_set_header Host umami.example.com;  # Here

        proxy_buffering on;

        add_header X-Cache $upstream_cache_status;
    }

    location = /js/api/send {
        proxy_pass $umami_event_url;
        proxy_set_header Host umami.example.com;  # Here
        proxy_buffering on;
        proxy_http_version 1.1;

        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Host $host;
    }
``` 

Then just modify your script tag.

```
<script async src="/js/script.js" data-website-id="dlsjfnjf-safasd65-f8we45f"></script>
```

Now your umami page views and button events are proxied via nginx.

**Comment by ForsakenRei** [comment]
Thanks for the config. I use nginx proxy manager instead of the old school nginx, so I put those configs to the custom config in the advanced tab, but it seems not working as expected. Any suggestions?
